sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamValuesRel.java

```diff
@@ -80,15 +80,13 @@ public BeamValuesRel(
           BeamValuesRel.class.getSimpleName(),
           pinput);
 
+      Schema schema = CalciteUtils.toSchema(getRowType());
       if (tuples.isEmpty()) {
-        throw new IllegalStateException("Values with empty tuples!");
+        return pinput.getPipeline().begin().apply(Create.empty(schema));
+      } else {
+        List<Row> rows = tuples.stream().map(tuple -> tupleToRow(schema, tuple)).collect(toList());
+        return pinput.getPipeline().begin().apply(Create.of(rows)).setRowSchema(schema);
```

Comment:
I think that if you only keep the `else` part and remove the `tuples.isEmpty()` block it should still work, `Create.of(..)` should work fine for `nulls` or empty args.

Reply:
You are right.